### DISCUSSION TITLE
api, get,  /api/realtime;  takes too long to return 5 minutes PB_CONNECT,

### DISCUSSION THREAD

**Discussion by ejrua** [pocketbase/pocketbase#741] [Q&A]
takes too long to return 5 minutes PB_CONNECT, please help

**Comment by ganigeorgiev** [comment]
I don't understand. Could you elaborate a little more on the issue?

`PB_CONNECT` is the first message that you get when you establish the SSE connection. 5 minutes is the idle timeout, aka. if you haven't received any messages, the connection will be dropped and an attempt to reconnect will be made in the background by the SDK automatically. This is an intended behavior to protect from "zombie/leaked" connections.

**Reply by ganigeorgiev** [reply]
After a second read, I guess you expect when you send `GET /api/realtime` to get immediate response, right? But this is not how [SSE](https://developer.mozilla.org/en-US/docs/Web/API/Server-sent_events) works. `GET /api/realtime` establishes a persistent HTTP connection and returns a **streamed response** that follows the SSE Message spec format, eg:
```
id: someid
event: create
data: {...}
\n\n -> indicates end of the message
```

If you use the JavaScript SDK, the connection and subscription is all handled automatically for you by just running:

```js
client.realtime.subscribe("YOUR_COLLECTION", function (e) {
    console.log(e.action);
    console.log(e.record);
});
```

**Reply by ejrua** [reply]
" I guess you expect when you send GET /api/realtime to get immediate response, right?", yes, SSE with python since there is no Client SDK, options?

**Reply by ganigeorgiev** [reply]
@ejrua You can use any SSE Python library. The general idea is the following:

1. Make a `GET /api/realtime` request to establish the connection. This is a persistent/long-living connection where you'll receive the event messages.

2. To subscribe, send a regular `POST /api/realtime` request with the clientId (received in the first PB_CONNECT event) and your subscriptions.

You can also find some info in https://github.com/pocketbase/pocketbase/discussions/308#discussioncomment-3389399.
